py_zipkin/zipkin.py

```diff
@@ -709,3 +713,125 @@ def create_http_headers_for_new_span(context_stack=None, tracer=None):
         'X-B3-Flags': '0',
         'X-B3-Sampled': '1' if zipkin_attrs.is_sampled else '0',
     }
+
+
+def is_valid_trace_id(candidate):
```

Comment:
Please remove validation or at least make it optional and disabled by default. Even though the spec says that ids should be 16 or 32 bytes long, the zipkin UI and api accepts also shorter ids and pads them with leading 0s.

This has unfortunately led to systems in our infrastructure using shorter ids, which wouldn't pass this validation.

Your call, but I'd just get rid of the validation which will also simplify your tests a lot

Reply:
No prob, ripping it out :)